actionview/lib/action_view/helpers/asset_url_helper.rb

```diff
@@ -159,7 +159,7 @@ def asset_path(source, options = {})
       #   asset_url "application.js", host: "http://cdn.example.com" # => http://cdn.example.com/assets/application.js
       #
       def asset_url(source, options = {})
-        path_to_asset(source, options.merge(:protocol => :request))
+        path_to_asset(source, options.merge(:protocol => respond_to?(:request) ? :request : config.default_asset_host_protocol || :relative))
```

Comment:
It would probably be clear to have something like:
```
protocol = if respond_to?(:request)
  :request
else
 config.default_asset_host_protocol || :relative
end
path_to_asset(source, options.merge(protocol: protocol))
```